etsin_finder/frontend/js/components/qvain/views/editor/submitButtonsV2.jsx

```diff
@@ -1,33 +1,88 @@
-import React, { useEffect } from 'react'
+import React, { useEffect, useState } from 'react'
+import styled from 'styled-components'
 import PropTypes, { instanceOf } from 'prop-types'
 import { observer } from 'mobx-react'
 import Translate from 'react-translate-component'
 import { useStores } from '../../utils/stores'
 import SubmitButton from './submitButton.styled'
+import TooltipHoverOnSave from '../../general/header/tooltipHoverOnSave'
 
 export const SubmitButtonsV2 = ({ submitButtonsRef, disabled, doiModal, history, idSuffix }) => {
   const {
     Qvain: {
-      Submit: { submitDraft, submitPublish },
+      Submit: {
+        submitDraft,
+        submitPublish,
+        draftValidationError,
+        publishValidationError,
+        prevalidate,
+        isDraftButtonDisabled,
+        isPublishButtonDisabled,
+      },
       original,
     },
     Env: { getQvainUrl },
   } = useStores()
 
+  const [draftButtonHover, setDraftButtonHover] = useState(false)
+  const [publishButtonHover, setPublishButtonHover] = useState(false)
+
   useEffect(() => {
+    prevalidate()
     if (original?.identifier) {
       history.replace(getQvainUrl(`/dataset/${original.identifier}`))
     }
-  }, [original, getQvainUrl, history])
+  }, [original, getQvainUrl, history, prevalidate])
+
+  const prepareErrors = (error = {}) => {
+    const { errors = [] } = error
+    return errors.map(err => `${err}\n`)
+  }
 
   return (
     <div ref={submitButtonsRef}>
-      <SubmitButton id={`draft-btn${idSuffix}`} disabled={disabled} onClick={submitDraft}>
-        <Translate content="qvain.saveDraft" />
-      </SubmitButton>
-      <SubmitButton id={`publish-btn${idSuffix}`} disabled={disabled} onClick={submitPublish}>
-        <Translate content="qvain.submit" />
-      </SubmitButton>
+      <TooltipHoverOnSave
+        isOpen={draftButtonHover}
+        errors={draftValidationError?.errors || []}
+        description="qvain.validationMessages.draft.description"
+      >
+        <WrapperDivForHovering
+          onMouseOver={() => {
```

Comment:
In this case I'd prefer onMouseEnter that doesn't trigger when you hover on the tooltip. Same for the publish button.